### PR TITLE
[TIMOB-15287] [TIMOB-16622] iOS: Fixed autoAdjustScrollViewInsets for search bars in UITableView and UIListView

### PR DESCRIPTION
**Pull request by cheekiatng**
Details: https://jira.appcelerator.org/browse/TIMOB-15287
Details: https://jira.appcelerator.org/browse/TIMOB-16622

### CODE REVIEW

File: iphone/Classes/TiUIListView.m [hunk 1]
```diff
@@ -417,7 +418,7 @@ -(void)buildResultsForSearchText
             NSUInteger maxItems = [[self.listViewProxy sectionForIndex:i] itemCount];
             for (int j = 0; j < maxItems; j++) {
                 NSIndexPath* thePath = [NSIndexPath indexPathForRow:j inSection:i];
-                id theValue = [self valueWithKey:@"searchableText" atIndexPath:thePath];
+                id theValue = [self valueWithKey:@"title" atIndexPath:thePath];
```
Comment (vishalduggal) [inline iphone/Classes/TiUIListView.m]: This should be `searchableText` which is the defined property for ListView

File: iphone/Classes/TiUIiOSNavWindowProxy.m [hunk 2]
```diff
@@ -87,14 +91,38 @@ -(id)tabGroup
 -(UINavigationController*)controller
 {
     if (navController == nil) {
-        navController = [[UINavigationController alloc] initWithRootViewController:[self rootController]];;
+		navController = [[UINavigationController alloc] initWithRootViewController:[self rootController]];
         navController.delegate = self;
         [TiUtils configureController:navController withObject:self];
-        if ([TiUtils isIOS7OrGreater]) {
-            [navController.interactivePopGestureRecognizer addTarget:self action:@selector(popGestureStateHandler:)];
```
Comment (vishalduggal) [inline iphone/Classes/TiUIiOSNavWindowProxy.m]: Why are we deleting this code?

File: iphone/Classes/TiUIiOSNavWindowProxy.m [hunk 3]
```diff
@@ -87,14 +91,38 @@ -(id)tabGroup
...
+			[navController.interactivePopGestureRecognizer addTarget:self action:@selector(popGestureStateHandler:)];
+			if ([self rootController].automaticallyAdjustsScrollViewInsets) {
+				if ([rootWindow.children count] > 0) {
+					for (TiViewProxy* childView in rootWindow.children ) {
+						if (childView != nil && [childView isKindOfClass:[TiUITableViewProxy class]]) {
+							TiUITableView *table = (TiUITableView *)[childView view];
+							if (table.tableController != nil) {
+								[[self rootController] addChildViewController:table.tableController];
+								[table.tableController didMoveToParentViewController:[self rootController]];
+								[[self rootController] setEdgesForExtendedLayout:UIRectEdgeNone];
```
Comment (vishalduggal) [inline iphone/Classes/TiUIiOSNavWindowProxy.m]: Why reset the `edgesForExtendedLayout`?

File: iphone/Classes/TiUIiOSNavWindowProxy.m [hunk 4]
```diff
@@ -87,14 +91,38 @@ -(id)tabGroup
...
+							}
+						}
+						if (childView != nil && [childView isKindOfClass:[TiUIListViewProxy class]]) {
+							TiUIListView *list = (TiUIListView *)[childView view];
+							if (list.tableController != nil) {
+								[[self rootController] addChildViewController:list.tableController];
+								[list.tableController didMoveToParentViewController:[self rootController]];
+								[[self rootController] setEdgesForExtendedLayout:UIRectEdgeNone];
```
Comment (vishalduggal) [inline iphone/Classes/TiUIiOSNavWindowProxy.m]: Same here. Why reset?

**Comment by vishalduggal** [comment]
The assumption being made here is that the tableView or listView has a frame where it lies underneath the navigation bar. What if it does not? 

Also there is no handling of the problem for TabGroups which also have a navigation controller. 

And what are we going to do if the tableView/listView is added to the window after is has opened or it is not a direct child of the WindowProxy?

The `autoAdjustScrollViewInsets` is only relevant when the ViewController->view is a ScrollView or a derived class. That is never the case in titanium where the view is always a plain UIView.

The only way to really fix this is to allow Tabs and NavigationWindows to present ViewControllers that are not managing WindowProxies ie `nav.openWindow(proxy)` and `tab.open(proxy)` now enforces the proxy object to be a WindowProxy. We should allow it to be any ViewProxy. Once that is done we can then modify the TiViewController to load the proxy->view as its view and go from there.

That change is not going to happen anytime soon since it will require significant refactor of the SDK and a sign off from other participating SDK components for the changes required. 

This PR is __REJECTED__

**Comment by hansemannn** [comment]
Closing PR

**Pull request closed without merging**